### DISCUSSION TITLE
Reload does not work with fastapi[all]==0.87.0 or greater

### DISCUSSION THREAD

**Discussion by LucasHartridge** [fastapi/fastapi#6089] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
## Project Structure
.
├── app
│   └── main.py
├── Dockerfile
└── requirements.txt
└── docker-compose.yml

------------------------------------------------
## main.py

from fastapi import FastAPI

app = FastAPI()


@app.get("/")
def get_root():
    return {"message": "I am root!"}

------------------------------------------------

## Dockerfile

FROM amazon/aws-sam-cli-build-image-python3.9

RUN mkdir -p /cool-service

# Set environment variables
ENV PYTHONUNBUFFERED=1
ENV PYTHONDONTWRITEBYTECODE=1
ENV PYTHONPATH="${PYTHONPATH}:/cool-service"


COPY . /cool-service

RUN pip install --no-cache-dir --upgrade -r /cool-service/requirements.txt
RUN pip install debugpy

WORKDIR /cool-service

------------------------------------------------

## docker-compose.yml

version: "3.8"

services:
  cool-service:
    platform: linux/amd64
    container_name: cool-service
    build:
      context: .
      dockerfile: Dockerfile
    image: cool-service
    volumes:
      - .:/cool-service
    ports:
      - 5000:5000
    environment:
      - DEBUG=1
    command: python -m uvicorn app.main:app --host 0.0.0.0 --port 5000 --reload --reload-dir app

------------------------------------------------
## requirements.txt

fastapi[all]==0.87.0
uvicorn==0.20.0
```

### Description

Hey team :wave: 

I am working on a more complex project where I have a docker-compose building multiple services for my local environment and I noticed that after updating fastapi to 0.87.0 the hot reload stopped working for me 

After updating my services to use Fast API 0.87.0. My Unit test started failing because starlette does not include httpx. I found a GitHub [issue](https://github.com/tiangolo/fastapi/issues/5656) it says to use the TestClient from FastAPI it is necessary to install all the dependencies fastapi[all]

So on my requirements/test.txt I have fastapi[all]==0.87.0 and this is used to build the docker images on the Docker Compose since on those requirements I also have other libs that help to debug but are not required when we promote our code to an environment

Is not a blocker, is something that I would like to post for this community since this can easily be solved by using fastapi==0.87.0 or fastapi==0.88.0 

I guess this issue, in the case is an issue, was introduced on version 0.87.0 based on the necessity to use fastapi[all] for TestClient

In the code example, I reproduced this issue with the minimum possible steps and I tried to follow https://fastapi.tiangolo.com/deployment/docker/ to make it even simpler

After setting all the project structure just run:
- Run on terminal `docker compose up --build`
- Go and update main.py and save it

**Actual Result:** You will notice that the image will not be refreshed with the changes on main.py

**Expected Results:** Image is refreshed with recent changes on main.py

### Operating System

Linux

### Operating System Details

Ubuntu 22.04

### FastAPI Version

0.87.0 | 0.88.0 

### Python Version

3.9

**Comment by jgould22** [comment, marked as answer]
Both of these work for me.

The only difference between the two for me is 
```
7c7
< fastapi==0.88.0
---
> fastapi==0.87.0
24c24
< starlette==0.22.0
---
> starlette==0.21.0
```

Docker is known to mess with watchfiles because of the indirection of the filesystem. Try setting `WATCHFILES_FORCE_POLLING=true` in your env vars and see if that fixes it for you. 

Otherwise I am not sure why it might not be working since I cannot reproduce it.

**Comment by LucasHartridge** [comment]
Hey @jgould22 

Thanks for the quick response

I set my Dockerfile to use ENV WATCHFILES_FORCE_POLLING=true and it works totally fine. 

Just for clarification, I was experiencing this issue when fastapi[all]==0.87.0 or fastapi[all]==0.88.0 was used when fastapi==0.87.0 or fastapi==0.88.0 was working all good

Also, I was using fastapi==0.85.1 before and without setting the WATCHFILES_FORCE_POLLING was working totally fine
